source/slang/core.meta.slang

```diff
@@ -1255,37 +1255,44 @@ struct ParameterBlock {}
 __generic<T, let MAX_VERTS : uint>
 __magic_type(VerticesType)
 __intrinsic_type($(kIROp_VerticesType))
-struct Vertices
+[__NonCopyableType]
+struct OutputVertices
 {
     __subscript(uint index) -> T
     {
-        // TODO: Ellie make sure these remains write only
-        __intrinsic_op($(kIROp_GetElementPtr))
+        // TODO: Make sure this remains write only, we can't do this with just
+        // a 'set' operation as it's legal to only write to part of the output
+        // buffer, or part of the output buffer at a time.
+        __intrinsic_op($(kIROp_MeshOutputRef))
         ref;
     }
 };
 
 __generic<T, let MAX_PRIMITIVES : uint>
 __magic_type(IndicesType)
 __intrinsic_type($(kIROp_IndicesType))
-struct Indices
+[__NonCopyableType]
+struct OutputIndices
 {
     __subscript(uint index) -> T
     {
-        // TODO: Ellie: It's illegal to not write out the whole primitive at once, should we use set over ref?
-        __intrinsic_op($(kIROp_GetElementPtr))
-        ref;
+        // It's illegal to not write out the entire primitive at once, so limit
+        // this to set
+        [mutating]
+        __intrinsic_op($(kIROp_MeshOutputSet))
```

Comment:
We maybe able to get rid of this opcode by having `set` call `__getMeshOutputRef() = newValue;`